pennylane/compiler/python_compiler/transforms/combine_global_phases.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains the implementation of the combine_global_phases transform,
+written using xDSL."""
+
+from dataclasses import dataclass
+
+from xdsl import context, passes, pattern_rewriter
+from xdsl.dialects import arith, builtin, func, scf
+from xdsl.dialects.func import ReturnOp
+from xdsl.dialects.scf import ForOp, IfOp, WhileOp
+from xdsl.rewriter import InsertPoint
+
+from ..quantum_dialect import GlobalPhaseOp
+from .utils import xdsl_transform
+
+
+class CombineGlobalPhasesPattern(
+    pattern_rewriter.RewritePattern
+):  # pylint: disable=too-few-public-methods
+    """RewritePattern for removing ~:class:`~pennylane.GlobalPhase` gates in the circuit (if exists) and adding
+    at the end of list of operations with its phase being a total global phase computed as the algebraic sum of
+    all global phases in the original circuit."""
+
+    # pylint: disable=no-self-use
+    @pattern_rewriter.op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, root: func.FuncOp, rewriter: pattern_rewriter.PatternRewriter
+    ):  # pylint: disable=arguments-differ
+        """Implementation of rewriting FuncOps that may contain operations corresponding to
+        GlobalPhase oprations."""
+        phi = None
+        global_phases = []
+        for region in root.regions:
+            for op in region.ops:
+                if isinstance(op, GlobalPhaseOp):
+                    global_phases.append(op)
+                elif isinstance(op, (IfOp, ForOp, WhileOp)):
+                    continue
+
+        if len(global_phases) < 2:
+            return
+
+        prev = global_phases[0]
+        phi_sum = prev.operands[0]
+        for current in global_phases[1:]:
+            phi = current.operands[0]
+            addOp = arith.AddfOp(phi, phi_sum)
+            rewriter.insert_op(addOp, InsertPoint.before(current))
+            phi_sum = addOp.result
+
+            rewriter.erase_op(prev)
+            prev = current
+
+        prev.operands[0] = phi_sum
```

Comment:
```suggestion
        prev.operands[0].replace_by_if(phi_sum, lambda use: use.operation == prev)
```

I think this should also work.